models/issues/issue.go

```diff
@@ -501,6 +503,41 @@ func GetIssueByIndex(ctx context.Context, repoID, index int64) (*Issue, error) {
 	return issue, nil
 }
 
+func isPullToCond(isPull optional.Option[bool]) builder.Cond {
+	if isPull.Has() {
+		return builder.Eq{"is_pull": isPull.Value()}
+	}
+	return builder.NewCond()
+}
+
+func FindLatestIssues(ctx context.Context, repoID int64, isPull optional.Option[bool], pageSize int) (IssueList, error) {
+	issues := make([]*Issue, 0, pageSize)
+	err := db.GetEngine(ctx).Where("repo_id = ?", repoID).
+		And(isPullToCond(isPull)).
+		OrderBy("created_unix DESC").
+		Limit(pageSize).
+		Find(&issues)
+	return issues, err
+}
+
+func FindIssuesSuggestionByKeyword(ctx context.Context, repoID int64, keyword string, isPull optional.Option[bool], pageSize int) (IssueList, error) {
+	indexKeyword, _ := strconv.ParseInt(keyword, 10, 64)
+	cond := builder.NewCond()
+	if indexKeyword > 0 {
+		cond = cond.Or(builder.Eq{"`index`": indexKeyword})
```

Comment:
I think we need a separate search to make the "number" be the first one.

Otherwise it might not appear due to `		Limit(pageSize).`

Reply:
e24690d596